src/app/map/components/GoogleMaps/index.tsx

```diff
@@ -1,87 +1,151 @@
-              zoom: 16,
-              styles: colorScheme === "dark" ? mapTheme.dark : mapTheme.light,
-            };
+      await loader.load();
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          const { latitude, longitude } = position.coords;
+          const mapOptions = {
+            center: { lat: latitude, lng: longitude },
+            zoom: 10,
+            styles: colorScheme === "dark" ? mapTheme.dark : mapTheme.light,
+          };
+          setCurrentMarker(
+            <Marker position={{ lat: latitude, lng: longitude }} />
+          );
 
-            const newMap = new window.google.maps.Map(
-              document.getElementById("map") as HTMLElement,
-              mapOptions
-            );
-
-            new window.google.maps.Marker({
-              position: { lat: latitude, lng: longitude },
-              map: newMap,
+          if (dummyData !== null) {
+            const markerElements = dummyData.map((event, i) => {
+              const index = i + 1;
+              return (
+                event?.venue?.lat &&
+                event?.venue?.lng && (
+                  <Marker
+                    key={index}
+                    position={{
+                      lat: event.venue.lat,
+                      lng: event.venue.lng,
+                    }}
+                    onClick={() => {
+                      setInfoWindowID(index);
+                    }}
+                    icon={{ url: "/images/marker.svg" }}
+                  >
+                    {infoWindowID === index && (
+                      <InfoWindow>
+                        <EventCard event={event} />
+                      </InfoWindow>
+                    )}
+                  </Marker>
+                )
+              );
             });
-
-            setMap(newMap);
-            setMapLoaded(true);
-          },
-          (error) => {
-            console.error("Error getting current location:", error);
-            const mapOptions = {
-              center: { lat: 49.2838, lng: -123.1193 },
-              zoom: 16,
-              styles: colorScheme === "dark" ? mapTheme.dark : mapTheme.light,
-            };
-            const newMap = new window.google.maps.Map(
-              document.getElementById("map") as HTMLElement,
-              mapOptions
-            );
-
-            setMap(newMap);
+            setMarkers(markerElements as JSX.Element[]);
           }
-        );
-      });
+
+          setMapLoaded(true);
+        },
+        (error) => {
+          console.error("Error getting current location:", error);
+        }
+      );
     };
 
     void fetchMap();
-  }, [colorScheme, setMapLoaded]);
+  }, [colorScheme, setMapLoaded, infoWindowID]);
 
   return (
-    <div className={classes.wrapper}>
-      <div id="map" className={classes.googleMap}></div>
+    <div>
+      <LoadScript
+        googleMapsApiKey="AIzaSyBno3t41t6dGC-Krh57KQbKU_giH9XQwRU"
```

Comment:
```suggestion
        googleMapsApiKey=process.env.NEXT_PUBLIC_GOOGLE_MAPS_KEY
```
It's a good habit to use environment variables to hide the actual keys in public repos.
It's quite dangerous to expose any keys with GitHub because we can't completely remove the commit history once it's pushed to the remove repo.
So we have no choice but change key values  if it's sensitive data such as a database password.

Reply:
Ah yeah, it's definitely a better practice to use environment variable! Thank you for pointing it out! I updated it :)